modules/nlu/src/backend/middleware.ts

```diff
@@ -0,0 +1,89 @@
+import 'bluebird-global'
+import * as sdk from 'botpress/sdk'
+import util from 'util'
+
+import ScopedEngine from './engine'
+
+export const registerMiddleware = async (bp: typeof sdk, botScopedNlu: EngineByBot) => {
+  bp.events.registerMiddleware({
+    name: 'nlu.incoming',
+    direction: 'incoming',
+    handler: async (event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) => {
+      await processEvent(event)
+      next()
+    },
+    order: 10,
+    description:
+      'Process natural language in the form of text. Structured data with an action and parameters for that action is injected in the incoming message event.'
+  })
+
+  async function processEvent(event) {
```

Comment:
is there a reason why this function is nested in the registerMiddleware function ? to me it just remove code readability

Reply:
nope I just didn't rewrite this class. Will do